src/executor.rs

```diff
@@ -25,161 +30,90 @@ pub struct Task {
     pub backtrace: bool,
 }
 
+type LineSender = crossbeam::channel::Sender<CommandExecInfo>;
+
 impl Executor {
     /// launch the commands, send the lines of its stderr on the
     /// line channel.
     /// If `with_stdout` capture and send also its stdout.
     pub fn new(mission: &Mission) -> Result<Self> {
-        let mut command = mission.get_command();
+        let mut command = Command::from(mission.get_command());
         let with_stdout = mission.need_stdout();
-        let (task_sender, task_receiver) = bounded::<Task>(1);
-        let (stop_sender, stop_receiver) = bounded(0);
-        let (line_sender, line_receiver) = unbounded();
+        let (task_sender, mut task_receiver) = channel::<Task>(1);
+        let (stop_sender, mut stop_receiver) = oneshot::channel();
```

Comment:
unclear where is handle the signal, and tokio have tool that already return a stream https://docs.rs/tokio/latest/tokio/signal/index.html

Reply:
ctrl+c is handled elsewhere `src/keybindings.rs` seems to define a handler for it (later used in `app.rs`). This "signal" is then used by the app in the `Executor::die` function to signal that the child process should be terminated.

Reply:
if possible I would use tokio to handle signal too avoid this extra channel would be nice

Reply:
This would require restructuring of the rest of bacon, which feels outside the scope of this PR.

Reply:
Unfortunately, your PR is a big change to bacon, introduce tokio/async is not a small thing it's a major change for bacon.